core/src/main/java/org/apache/carbondata/core/datastore/compression/Compressor.java

```diff
@@ -18,36 +18,39 @@
 package org.apache.carbondata.core.datastore.compression;
 
 import java.io.IOException;
+import java.nio.ByteBuffer;
 
 public interface Compressor {
 
   String getName();
 
-  byte[] compressByte(byte[] unCompInput);
+  ByteBuffer compressByte(ByteBuffer compInput);
```

Comment:
Similar problem exists in query flow also ? can we use `ByteBuffer` for `uncompress` return type also ?

Reply:
Many places need to change for query flow, can not do in this PR